onnxruntime/core/providers/nv_tensorrt_rtx/nv_provider_factory.cc

```diff
@@ -516,32 +519,481 @@ struct NvTrtRtxSyncStreamImpl : OrtSyncStreamImpl {
+      cuDestroyExternalMemory(ext_memory);
+      return impl.ort_api.CreateStatus(ORT_FAIL, "Failed to allocate external memory handle");
+    }
+
+    handle->ext_memory = ext_memory;
+    handle->mapped_ptr = mapped_ptr;
+    handle->size_bytes = desc->size_bytes;
+    handle->offset_bytes = desc->offset_bytes;
+    handle->handle_type = desc->handle_type;
+    handle->is_dedicated = is_dedicated;
+
+    *out_handle = reinterpret_cast<OrtExternalMemoryHandleImpl*>(handle);
+    return nullptr;
+  }
+
+  static void ORT_API_CALL ReleaseMemoryImpl(
+      _In_ OrtExternalResourceImporterImpl* this_ptr,
+      _In_ OrtExternalMemoryHandleImpl* handle) noexcept {
+    auto& impl = *static_cast<NvTrtRtxExternalResourceImporterImpl*>(this_ptr);
+    (void)impl;
+
+    if (handle == nullptr) {
+      return;
+    }
+
+    auto* mem_handle = reinterpret_cast<NvTrtRtxExternalMemoryHandle*>(handle);
+
+    // Destroy the external memory object (also releases mapped buffer)
+    if (mem_handle->ext_memory != nullptr) {
+      cuDestroyExternalMemory(mem_handle->ext_memory);
+    }
+
+    delete mem_handle;
+  }
+
+  static OrtStatus* ORT_API_CALL CreateTensorFromMemoryImpl(
+      _In_ OrtExternalResourceImporterImpl* this_ptr,
+      _In_ const OrtExternalMemoryHandleImpl* mem_handle,
+      _In_ const OrtExternalTensorDescriptor* tensor_desc,
+      _Outptr_ OrtValue** out_tensor) noexcept {
+    auto& impl = *static_cast<NvTrtRtxExternalResourceImporterImpl*>(this_ptr);
+
+    if (mem_handle == nullptr || tensor_desc == nullptr || out_tensor == nullptr) {
+      return impl.ort_api.CreateStatus(ORT_INVALID_ARGUMENT, "Invalid arguments to CreateTensorFromMemory");
+    }
+
+    // Validate descriptor version
+    if (tensor_desc->version != ORT_EXTERNAL_TENSOR_DESCRIPTOR_VERSION) {
+      return impl.ort_api.CreateStatus(ORT_INVALID_ARGUMENT,
+                                       "Invalid OrtExternalTensorDescriptor version");
+    }
+
+    *out_tensor = nullptr;
+
+    auto* handle = reinterpret_cast<const NvTrtRtxExternalMemoryHandle*>(mem_handle);
+
+    // Validate tensor offset does not exceed available buffer size
+    size_t available_size = handle->size_bytes - handle->offset_bytes;
+    if (tensor_desc->offset_bytes > available_size) {
+      return impl.ort_api.CreateStatus(ORT_INVALID_ARGUMENT,
+                                       "tensor offset_bytes exceeds available imported memory size");
+    }
+
+    // Calculate the data pointer with tensor offset
+    void* data_ptr = reinterpret_cast<void*>(handle->mapped_ptr + tensor_desc->offset_bytes);
+
+    // Create memory info for CUDA device
+    OrtMemoryInfo* memory_info = nullptr;
+    OrtStatus* status = impl.ort_api.CreateMemoryInfo_V2(
+        "NvTensorRTRTX",
+        OrtMemoryInfoDeviceType_GPU,
+        OrtDevice::VendorIds::NVIDIA,
+        impl.device_id_,
+        OrtDeviceMemoryType_DEFAULT,
+        0,  // alignment
+        OrtDeviceAllocator,
+        &memory_info);
```

Comment:
Ideally this should come from the OrtEpDevice. That's available to an API user via EpDevice_MemoryInfo.